### ISSUE TITLE
[Bug]: Dragging a Document Link onto itself breaks the document tree

### ISSUE THREAD

**Issue by ThisIsJustARandomGuy** [pimcore/pimcore#12599]
### Expected behavior

Pimcore continues to not be broken. It shouldn't be possible for a link to link to itself.

### Actual behavior

The Document tree won't expand the parent node of the link and gets stuck in an endless loop.

### Steps to reproduce

1. Right click on any node in the document tree and "Add Link". /testing-links in this example:
![image](https://user-images.githubusercontent.com/21229739/177512165-c84102d8-f624-4fd6-acbb-837ce10b3e57.png)
2. Name the link, open and "Save & Publish" it. /testing-links/test in this example:
![image](https://user-images.githubusercontent.com/21229739/177513967-43d2d554-d7c0-45f2-9830-f107dcfa68a2.png)
3. Take the same link, drag it onto itself and "Save & Publish" again.
![image](https://user-images.githubusercontent.com/21229739/177514165-f9f0f9a8-3968-40b3-8127-7a9ff66827a1.png)
5. The document tree is now stuck loading and won't expand /testing-links. Eventually, it fails with an out of memory exception.
![image](https://user-images.githubusercontent.com/21229739/177514559-3e647a49-18a6-4232-bb1a-b83c2dfdc5c4.png)

Deleting the parent node also does not work for the same reason, I assume. A quick and easy fix is to either manually delete the database entry of the link in the `documents` table or change its target ID in the `documents_link` table.

It's a pretty easy thing to mess up on accident, so I think there should be some protection.

Tested on two different installs of 10.4.4

**Comment by ThisIsJustARandomGuy** [comment]
For anyone running into this: As a quick set-and-forget workaround I created an event subscriber. I'm sure that a better solution would be a quick check in `pimcore/bundles/AdminBundle/Resources/public/js/pimcore/document/link.js` though.

First make sure, that the EventSubscriber is tagged correctly. In your `services.yaml` there should be this or an equivalent:

```yaml
services:
    App\EventSubscriber\:
        resource: '../src/EventSubscriber/*'
        tags: [ 'kernel.event_subscriber' ]
```

Then just create the file `src/EventSubscriber/LinkSaveSubscriber.php`:

```PHP
<?php
namespace App\EventSubscriber;

use Pimcore\Event\DocumentEvents;
use Pimcore\Event\Model\DocumentEvent;
use Pimcore\Model\Document\Link;
use Pimcore\Model\Element\ValidationException;
use Symfony\Component\EventDispatcher\EventSubscriberInterface;

class LinkSaveSubscriber implements EventSubscriberInterface
{
    public static function getSubscribedEvents(): array
    {
        return [
            DocumentEvents::POST_ADD => 'onPostUpdate',
            DocumentEvents::POST_UPDATE => 'onPostUpdate',
        ];
    }

    public function onPostUpdate(DocumentEvent $event): void
    {
        $link = $event->getDocument();

        // Skip if not a link to a document
        if( !$link instanceof Link || $link->getType() !== 'link' || $link->getLinktype() !== 'internal' || $link->getInternalType() !== 'document') {
            return;
        }

        // Check if the link links to itself. If it does, reset the target
        // and throw a validation exception to inform the user
        if( $link->getId() === $link->getInternal() ) {
            $link->setElement(null);
            $link->setInternalType(null);
            $link->save();

            throw new ValidationException('Link to itself is not allowed and was reset. Please reload Pimcore.');
        }
    }
}
```

**Comment by dvesh3** [comment]
Fixed by #12625
